app/com/linkedin/drelephant/spark/heuristics/ConfigurationHeuristic.scala

```diff
@@ -128,18 +135,29 @@ object ConfigurationHeuristic {
     lazy val serializer: Option[String] = getProperty(SPARK_SERIALIZER_KEY)
 
     lazy val serializerSeverity: Severity = serializer match {
-      case None => Severity.NONE
+      case None => Severity.MODERATE
       case Some(`serializerIfNonNullRecommendation`) => Severity.NONE
-      case Some(_) => serializerIfNonNullSeverityIfRecommendationUnmet
+      case Some(_) => DEFAULT_SERIALIZER_IF_NON_NULL_SEVERITY_IF_RECOMMENDATION_UNMET
+    }
+
+    lazy val isDynamicAllocationEnabled: Option[Boolean] = getProperty(SPARK_DYNAMIC_ALLOCATION_ENABLED).map(_.toBoolean)
```

Comment:
Is there a reason for not applying defaults at declaration site? I might be wrong, but it looks like in the following match statement you treat `None` as `false` for both properties. 

Here's a rewritten match:

```scala
 lazy val shuffleAndDynamicAllocationSeverity = (isDynamicAllocationEnabled, isShuffleServiceEnabled) match {
   case (_, Some(true)) | (None, Some(true)) => Severity.NONE
   case (Some(true), Some(false)) | (Some(true), None) => Severity.SEVERE
   case (Some(false), Some(false)) => Severity.MODERATE
   case (Some(false), None) => Severity.MODERATE
   case (None, Some(false)) => Severity.MODERATE
   case (None, None) => Severity.MODERATE
 }
```

The first case is essentially 

```scala
case (_, true) => Severity.None
```

the next bunch is

```scala
case (true, false) => Severity.SEVERE
```

and finally

```scala
case (false, false) => Severity.MODERATE
```